packages/core/src/utils/stripTypename.ts

```diff
@@ -0,0 +1,21 @@
+export const stripTypename = (data: any): any => {
+  if (!data || typeof data !== 'object') return data;
+
+  return Object.keys(data).reduce((acc, key: string) => {
+    const value = data[key];
+    if (key === '__typename') {
+      Object.defineProperty(acc, '__typename', {
+        enumerable: false,
+        value,
+      });
+    } else if (Array.isArray(value)) {
+      acc[key] = value.map(stripTypename);
+    } else if (typeof value === 'object') {
```

Comment:
This needs an escape hatch:

```suggestion
    } else if (typeof value === 'object' && '__typename' in value) {
```

That's because we don't know what other values may be used here. It's imaginable that a user has deserialised a result at this point where a value for instance is a `Date` object.